lib/auth_app/accounts.ex

```diff
@@ -234,6 +261,60 @@ defmodule AuthApp.Accounts do
     Repo.one(query)
   end
 
+  @doc """
+  Gets the user with the given magic link token.
+  """
+  def get_user_by_magic_link_token(token) do
+    with {:ok, query} <- UserToken.verify_magic_link_token_query(token),
+         {user, _token} <- Repo.one(query) do
+      user
+    else
+      _ -> nil
+    end
+  end
+
+  @doc """
+  Gets the user with the given signed token.
+  """
+  def magic_link_sign_in(token) do
+    {:ok, query} = UserToken.verify_magic_link_token_query(token)
+
+    case Repo.one(query) do
+      # TODO: check if we really want to mix confirmation and login
+      {%User{email: current_email, confirmed_at: nil} = user, %UserToken{sent_to: current_email}} ->
+        {:ok, %{user: user, tokens_to_expire: tokens_to_expire}} =
+          Repo.transaction(confirm_user_multi(user))
```

Comment:
We don't need to delete previous tokens in this case, no tokens should exist. We should delete only the current one.

Reply:
Yes, but I wonder: what if a user changes the system to allow being logged in without a confirmed account? Then it would be good to invalidate all existing tokens.

Reply:
I think allowing unconfirmed access is a bad practice we don't want to encourage anyway, especially if we start with magic links only. So I'd not worry about this :)

Reply:
@SteffenDE based on the discussion below, we want to keep this in. Both confirming the account and changing the password signs everyone out. We should probably make it so changing the email signs everyone out too, this way we can just share this logic across everywhere?

Reply:
I began to implement this, but then reverted because we currently change the email directly in the settings LiveView and if we want to do it like the password change, we'd need to move it into the session controller to reauthenticate or mix changing email into the magic link route as well. The latter one would complicate the magic link conditions even more, as we'd need to check the token context for change email - so I'm not sure if that will really make things easier.